keras/preprocessing/image.py

```diff
@@ -771,7 +771,9 @@ def __init__(self, x, y, image_data_generator,
 
         if data_format is None:
             data_format = K.image_data_format()
-        self.x = np.asarray(x, dtype=K.floatx())
+        if not getattr(x, 'dtype', None) != K.floatx():
```

Comment:
`if not {expression} != {expression}:` is quite a strange structure. Did you mean:

```python
dtype = getattr(x, 'dtype', None)
if dtype != K.floatx():
   x = np.asarray(x, dtype=K.floatx())
```

Reply:
Yeah that's looks nicer, I'll change it